ai/src/main/java/org/triplea/ai/flowfield/map/FieldTerritory.java

```diff
@@ -0,0 +1,34 @@
+package org.triplea.ai.flowfield.map;
+
+import games.strategy.engine.data.Territory;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.ToString;
+
+@Data
+@EqualsAndHashCode(exclude = "neighbors")
```

Comment:
`neighbors` needs to be excluded because otherwise lombok will create an equals and hashcode that recurses n itself.

Reply:
This is one pitfall of lombok, I would suggest to document this in a test-case, by doing a quick test-case that sets up a minimal `FieldTerritory`, adds data to the list that would recurse and doing an `equals` call. In case someone were to try and be clever and remove this exclude, or somehow trigger it otherwise, we'd have the test case to catch it.